server/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -261,6 +261,13 @@ public void set(long index, byte[] buf, int offset, int len) {
             assert index >= 0 && index < size();
             System.arraycopy(buf, offset << 3, array, (int) index << 3, len << 3);
         }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            int size = (int) size();
+            out.writeVInt(size * 8);
```

Comment:
Maybe just cache `size * Long.BYTES` to a variable? :) Calculating it twice in two different ways back to back makes me unhappy :D